src/Servers.jl

```diff
@@ -60,18 +64,18 @@ function check_rate_limit(tcp, rate_limit::Rational{Int})
 end
 
 "Convenience object for passing around server details"
-struct Server2{S, I}
-    ssl::S # Union{SSLConfig, Nothing}; Nothing if non-SSL
+struct Server{S <: Union{SSLConfig, Nothing}, I <: Base.IOServer}
```

Comment:
My only concern here was if people had code calling `HTTP.Server` and now it's changed. But I guess that'll be an issue anyway, so I guess this is fine.

Reply:
But, this `HTTP.Servers.Server` is not exported right?